packages/components/src/composite/legacy/stories/utils.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import type { StoryContext } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import type { LegacyStateOptions } from '..';
+
+export function UseCompositeStatePlaceholder( props: LegacyStateOptions ) {
+	return (
+		<dl>
+			{ Object.entries( props ).map( ( [ name, value ] ) => (
+				<>
+					<dt>{ name }</dt>
+					<dd>{ JSON.stringify( value ) }</dd>
+				</>
+			) ) }
+		</dl>
+	);
+}
+UseCompositeStatePlaceholder.displayName = 'useCompositeState';
+
+export function transform( code: string, context: StoryContext ) {
```

Comment:
There's a lot of overlap with the other `transform()` util. Does it make sense to unify and avoid repetition?